src/lib/navigation/configurable-navigation-mode.ts

```diff
@@ -0,0 +1,107 @@
+import {BaseNavigationMode} from './base-navigation-mode.interface';
+import {WizardComponent} from '../components/wizard.component';
+import {WizardCompletionStep} from '../util/wizard-completion-step.interface';
+
+export class ConfigurableNavigationMode extends BaseNavigationMode {
+
+  constructor(
+    private navigateBackward: 'allow'|'deny'|null = null,
```

Comment:
Do we require the `= null` here?

Reply:
After looking at the line again I think I can answer that question myself: yes we do, because `'allow'|'deny'|null` doesn't allow for `undefined`

Reply:
No, it is not the right answer :)

You are not using TypeScript in strict mode (not using `strict` or, more specifically, `strictNullChecks` [compiler option](https://www.typescriptlang.org/docs/handbook/compiler-options.html)  in `tsconfig.json`). So, in angular-archwizard's codebase, `undefined` is assignable to `null`. Not only that, you would also be able to assign either `null` or `undefined` to a variable declared as simply `'allow'|'deny'`. I have included `|null` into the type signature 1) to be explicit; 2) because I am used to strict mode.

So, if we are not concerned with being strict, `private navigateBackward: 'allow'|'deny'|null = null` could be alternatively declared as `private navigateBackward?: 'allow'|'deny'`, the only difference being that the default value would be `undefined` rather than `null`.

Reply:
About `strict` mode: I've nothing about changing the `angular-archwizard` code base to use `strict` mode if we are able to. I'm always in favor of preventing bugs, which I think we can achieve by using `strict` mode. So if you think it is a good idea to change the project to use `strict` mode feel free to do so!

Reply:
Yes, I welcome the idea to switch to `strict` mode, but it can wait till another time.